cluster-autoscaler/simulator/node_info_utils_test.go

```diff
@@ -502,15 +510,27 @@ func TestCreateSanitizedNodeInfo(t *testing.T) {
 			taintConfig: &taintConfig,
 			wantTaints:  []apiv1.Taint{{Key: "a", Value: "b", Effect: apiv1.TaintEffectNoSchedule}},
 		},
-	} {
+		{
+			testName:                    "sanitize node with NodeDeclaredFeatures enabled",
+			nodeInfo:                    framework.NewNodeInfo(nodeWithDeclaredFeatures, resourceSlices, framework.NewPodInfo(pod1WithClaims, pod1ResourceClaims), framework.NewPodInfo(pod2WithClaims, pod2ResourceClaims)),
+			nodeDeclaredFeaturesEnabled: true,
+		},
+		{
+			testName:                    "sanitize node with NodeDeclaredFeatures disabled",
```

Comment:
is this update to `TestSanitizedTemplateNodeInfoFromNodeInfo` actually testing this same input+output as well (node w/ declared feature data but w/ feature gate disabled)?